cohttp/src/request.ml

```diff
@@ -77,7 +78,12 @@ let make ?(meth = `GET) ?(version = `HTTP_1_1) ?encoding ?headers uri =
   in
   let scheme = Uri.scheme uri in
   let resource = Uri.path_and_query uri in
-  Http.Request.make ~meth ~version ~headers ?scheme ?encoding resource
+  let headers =
+    match Header.get_transfer_encoding headers with
+    | Transfer.(Chunked | Fixed _) -> headers
+    | Unknown -> Header.add_transfer_encoding headers encoding
```

Comment:
I kept `Cohttp.{Request,Response}.make` backward compatible, @rgrinberg do you reckon we should just drop encoding also from here?